core/network/fabric-interop-cc/contracts/interop/handle_external_request.go

```diff
@@ -29,47 +30,75 @@ import (
 // 3. Checks the access control policy for the requester and view address is met
 // 4. Calls application chaincode
 func (s *SmartContract) HandleExternalRequest(ctx contractapi.TransactionContextInterface, b64QueryBytes string) (string, error) {
-	// Ensure that this function cannot be called by a client without relay permissions
-	relayAccessCheck, err := wutils.IsClientRelay(ctx.GetStub())
+	queryBytes, err := base64.StdEncoding.DecodeString(b64QueryBytes)
 	if err != nil {
-		return "", err
+		return "", logThenErrorf("Unable to base64 decode data: %s", err.Error())
 	}
-	if !relayAccessCheck {
-		return "", fmt.Errorf("Illegal access by relay")
+	var query common.Query
+	err = protoV2.Unmarshal(queryBytes, &query)
+	if err != nil {
+		return "", logThenErrorf("Unable to unmarshal query: %s", err.Error())
 	}
-	fmt.Println("Relay access check passed")
+	resp, err := handleRequest(s, ctx, query, query.Address)
+	return resp, err
+}
 
+func (s *SmartContract) HandleEventRequest(ctx contractapi.TransactionContextInterface, b64QueryBytes string, dynamicQueryArg string) (string, error) {
 	queryBytes, err := base64.StdEncoding.DecodeString(b64QueryBytes)
 	if err != nil {
-		errorMessage := fmt.Sprintf("Unable to base64 decode data: %s", err.Error())
-		log.Error(errorMessage)
-		return "", errors.New(errorMessage)
+		return "", logThenErrorf("Unable to base64 decode data: %s", err.Error())
 	}
 	var query common.Query
 	err = protoV2.Unmarshal(queryBytes, &query)
 	if err != nil {
-		errorMessage := fmt.Sprintf("Unable to unmarshal query: %s", err.Error())
-		log.Error(errorMessage)
-		return "", errors.New(errorMessage)
+		return "", logThenErrorf("Unable to unmarshal query: %s", err.Error())
 	}
+
+	queryAddress := query.Address
+	dynamicArgCount := strings.Count(query.Address, ":?")
+	if dynamicArgCount > 1 {
+		return "", logThenErrorf("Expected 1 dynamic argument in the event query address, but found %d", dynamicArgCount)
+	} else if dynamicArgCount == 1 {
+		queryAddress = strings.Replace(query.Address, ":?", ":"+dynamicQueryArg, 1)
```

Comment:
Add a TODO comment above this saying that this is stopgap logic and will be reviewed and possibly changed later.